components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -542,23 +544,33 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
          ! spatially variable baroclinic phase speed, the mode can be specified by
          ! config_GM_spatially_variable_baroclinic_mode
          if (local_config_GM_lat_variable_c2) then
+            lt1 = 0.0_RKIND
+            lt2 = 0.0_RKIND
             !$omp parallel
-            !$omp do schedule(runtime) private(k, cell1, cell2, sumN2, lt1, lt2)
+            !$omp do schedule(runtime) private(k, cell1, cell2, sumN2, lt1, lt2, ltSum, c_min)
             do iEdge = 1, nEdges
                cell1 = cellsOnEdge(1, iEdge)
                cell2 = cellsOnEdge(2, iEdge)
-               sumN2 = 0.0
-
+               sumN2 = 0.0_RKIND
+               ltSum = epsGM
                do k = minLevelEdgeBot(iEdge)+1, maxLevelEdgeTop(iEdge)-1
 
                   lt1 = 0.5_RKIND*(layerThickness(k,cell1) + layerThickness(k-1,cell1))
                   lt2 = 0.5_RKIND*(layerThickness(k,cell2) + layerThickness(k-1,cell2))
                   sumN2 = sumN2 + 0.5_RKIND*(lt1*sqrt(max(BruntVaisalaFreqTop(k,cell1),0.0_RKIND)) + &
                                 lt2*sqrt(max(BruntVaisalaFreqTop(k,cell2),0.0_RKIND)))
 
+                  ltSum = ltSum + 0.5*(lt1+lt2)
                end do
 
-               cGMphaseSpeed(iEdge) = max(config_GM_constant_gravWaveSpeed,                          &
+               ! Compute the minimum allowed speed of the first baroclinic mode. 
+               ! If config_GM_minBclModeSpeed_method='constant' use local_config_GM_constant_bclModeSpeed.
+               ! If config_GM_minBclModeSpeed_method='computed' use Brunt-Vaisala frequency on this edge.
+               ! See initialization of these variables in this modules init routine.
+               c_min = gm_minBclModeSpeed_constant + gm_minBclModeSpeed_compute_on*max(0.01_RKIND,sumN2/ltSum*(0.5*(lt1+lt2)))
+
+               ! Compute the speed of the first baroclinic mode from the Brunt-Vaisala frequency.
+               cGMphaseSpeed(iEdge) = max(c_min,                          &
                                           sumN2/(config_GM_spatially_variable_baroclinic_mode*3.141592_RKIND))
```

Comment:
@vanroekel I changed the flag names to 
```
    config_GM_constant_bclModeSpeed = 0.3
    config_GM_minBclModeSpeed_method = 'constant'
```
The only substantial change is the line above, but I use fixed coefficient values set on init to avoid if statements inside the loop.